pkg/common/cns-lib/volume/util.go

```diff
@@ -66,11 +75,49 @@ func IsDiskAttached(ctx context.Context, vm *cnsvsphere.VirtualMachine, volumeID
 	return "", nil
 }
 
+func hasNVMEController(ctx context.Context, devices object.VirtualDeviceList) bool {
+	log := logger.GetLogger(ctx)
+	_, err := devices.FindNVMEController("")
```

Comment:
`devices.FindNVMEController` may return nil if no NVMEController is available with the free slots.
For the last disk attached to NVME controller, if we get a fault and call this function, we will get nil, so we will not be able to return success for already attached disk to NVME controller.


`devices.FindNVMEController` is internally using `PickController`

```golang
	c := l.PickController((*types.VirtualNVMEController)(nil))
	if c == nil {
		return nil, errors.New("no available NVME controller")
	}
```

```golang
// PickController returns a controller of the given type(s).
// If no controllers are found or have no available slots, then nil is returned.
func (l VirtualDeviceList) PickController(kind types.BaseVirtualController) types.BaseVirtualController {
```

Reply:
I removed this function call: `FindNVMEController`. We need to check the controllerKey on the disk and see if the controllerKey matches with the NVMe controller key